src/Servers/Kestrel/shared/test/HttpParsingData.cs

```diff
@@ -364,13 +383,19 @@ public static IEnumerable<string> TargetWithNullCharData
             "8charact",
         };
 
-        public static IEnumerable<object[]> RequestHeaderInvalidData => new[]
+        public static IEnumerable<object[]> RequestHeaderInvalidDataLineFeedTerminator => new[]
         {
             // Missing CR
             new[] { "Header: value\n\r\n", CoreStrings.FormatBadRequest_InvalidRequestHeader_Detail(@"Header: value\x0A") },
             new[] { "Header-1: value1\nHeader-2: value2\r\n\r\n", CoreStrings.FormatBadRequest_InvalidRequestHeader_Detail(@"Header-1: value1\x0A") },
             new[] { "Header-1: value1\r\nHeader-2: value2\n\r\n", CoreStrings.FormatBadRequest_InvalidRequestHeader_Detail(@"Header-2: value2\x0A") },
 
+            // Missing colon
+            // new[] { "\n", CoreStrings.FormatBadRequest_InvalidRequestHeader_Detail(@"\x0A") },
```

Comment:
I am uncertain with this one, whether it should be valid or not in this context even for quirk mode.

Reply:
An empty line (two CRLF's) marks the end of the headers. This input seems incomplete.

Reply:
Agreed. `\n\n` should end the request headers. So `\n`, is valid but not a header.

I guess this means that `\r\n\n` should also end the request headers? It doesn't look like we're doing anything to prevent `\r\n` and `\n` from being mixed in a single set of request headers.